Framework/Geometry/src/Objects/Object.cpp

```diff
@@ -1644,7 +1644,23 @@ void Object::calcBoundingBoxByGeometry() {
       maxZ = std::max(maxZ, vector.Z());
     }
   } break;
-
+  case 2: // HEXAHEDRON
+  {
+    // Vectors are in the same order as the following webpage:
+    // http://docs.mantidproject.org/nightly/concepts/HowToDefineGeometricShape.html#hexahedron
+    auto &lf = vectors[1];
+    auto &lb = vectors[0];
+    auto &rb = vectors[3];
+    auto &rf = vectors[2];
+    auto dz = vectors[4] - lf;
+
+    minX = std::min(lf.X(), lb.X());
+    maxX = std::max(rb.X(), rf.X());
+    minY = lb.Y();
+    maxY = rf.Y();
+    minZ = 0;
+    maxZ = dz.Z();
```

Comment:
This implies that the bottom and top plane are always parallel to the x-y plane?